web/components/issues/issue-detail/sidebar.tsx

```diff
@@ -185,248 +185,278 @@ export const IssueDetailsSidebar: React.FC<Props> = observer((props) => {
-                  <div className="flex flex-wrap items-center py-2">
-                    <div className="flex items-center gap-x-2 text-sm text-custom-text-200 sm:basis-1/2">
-                      <LayoutPanelTop className="h-4 w-4 flex-shrink-0" />
-                      <p>Parent</p>
-                    </div>
-                    <div className="sm:basis-1/2">
-                      <IssueParentSelect
-                        workspaceSlug={workspaceSlug}
-                        projectId={projectId}
-                        issueId={issueId}
-                        issueOperations={issueOperations}
-                        disabled={!is_editable}
-                      />
-                    </div>
+            {(fieldsToShow.includes("all") || fieldsToShow.includes("estimate")) &&
+              areEstimatesEnabledForCurrentProject && (
+                <div className="flex items-center gap-2 h-8">
+                  <div className="flex items-center gap-1 w-2/5 flex-shrink-0 text-sm text-custom-text-300">
+                    <Triangle className="h-4 w-4 flex-shrink-0" />
+                    <span>Estimate</span>
                   </div>
-                )}
-
-                {(fieldsToShow.includes("all") || fieldsToShow.includes("blocker")) && (
-                  <IssueRelationSelect
-                    workspaceSlug={workspaceSlug}
-                    projectId={projectId}
-                    issueId={issueId}
-                    relationKey="blocking"
-                    disabled={!is_editable}
-                  />
-                )}
-
-                {(fieldsToShow.includes("all") || fieldsToShow.includes("blocked")) && (
-                  <IssueRelationSelect
-                    workspaceSlug={workspaceSlug}
+                  <EstimateDropdown
+                    value={issue?.estimate_point !== null ? issue.estimate_point : null}
+                    onChange={(val) =>
+                      issueOperations.update(workspaceSlug, projectId, issueId, { estimate_point: val })
+                    }
                     projectId={projectId}
-                    issueId={issueId}
-                    relationKey="blocked_by"
                     disabled={!is_editable}
+                    buttonVariant="transparent-with-text"
+                    className="w-3/5 flex-grow group"
+                    buttonContainerClassName="w-full text-left"
+                    buttonClassName={`text-sm ${issue?.estimate_point !== null ? "" : "text-custom-text-400"}`}
+                    placeholder="None"
+                    hideIcon
+                    dropdownArrow
+                    dropdownArrowClassName="h-3.5 w-3.5 hidden group-hover:inline"
                   />
-                )}
+                </div>
+              )}
+
+            {(fieldsToShow.includes("all") || fieldsToShow.includes("module")) && projectDetails?.module_view && (
+              <div className="flex items-center gap-2 h-8">
```

Comment:
This has to be refactored to create a common component and reuse rather than adding conditions